src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/ReleaseFileServiceTests.cs

```diff
@@ -392,15 +416,134 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
                 using var zip = ZipFile.OpenRead(path);
 
                 // Entries are sorted alphabetically
-                Assert.Equal(2, zip.Entries.Count);
+                Assert.Equal(3, zip.Entries.Count);
                 Assert.Equal("ancillary/ancillary.pdf", zip.Entries[0].FullName);
                 Assert.Equal("Test ancillary blob", zip.Entries[0].Open().ReadToEnd());
 
                 Assert.Equal("data/data.csv", zip.Entries[1].FullName);
                 Assert.Equal("Test data blob", zip.Entries[1].Open().ReadToEnd());
+
+                // Data guidance is generated if there is at least one data file
+                Assert.Equal("data-guidance/data-guidance.txt", zip.Entries[2].FullName);
+                Assert.Equal("Test data guidance blob", zip.Entries[2].Open().ReadToEnd());
             }
 
-            MockUtils.VerifyAllMocks(blobStorageService);
+            MockUtils.VerifyAllMocks(blobStorageService, dataGuidanceFileWriter);
```

Comment:
I was saying to Ben the other day, that I've started to put `VerifyAllMocks()` calls directly after the `var result = service.blah(...)` line and before the `result.AssertRight()`, the reason being that if `result` is actually a failure and that failure is down to a mock invocation failing to be called, you'll not know about it as the test'll fail at the `AssertRight` leaving you to have to debug the code or whatever to work out why.

Putting the Verify right after the service call will let you know if it's down to a mock failing to be called straight away.

Reply:
Fair enough, Will move these to after the method invocation!